plugin/rpcplugin/supervisor_test.go

```diff
@@ -56,6 +56,21 @@ func TestSupervisor_InvalidExecutablePath(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestSupervisor_NonExistentExecutablePath(t *testing.T) {
+	dir, err := ioutil.TempDir("", "")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+
+	ioutil.WriteFile(filepath.Join(dir, "plugin.json"), []byte(`{"id": "foo", "backend": {"executable": "thisfileshouldnotexist"}}`), 0600)
+
+	bundle := model.BundleInfoForPath(dir)
+	supervisor, err := SupervisorProvider(bundle)
+	require.NotNil(t, supervisor)
+	require.NoError(t, err)
+
+	require.Error(t, supervisor.Start())
```

Comment:
Which error occurs? I guess you can decide if it's important to differentiate or be specific. 👍